score-xml/src/main/java/io/cloudslang/content/xml/utils/XmlUtils.java

```diff
@@ -0,0 +1,207 @@
+            }
+            node.getParentNode().insertBefore(beforeNode.cloneNode(true), node);
+        }
+    }
+
+    public static void removeFromList(NodeList nodeList, String attributeName) throws Exception{
+        for (int i = 0; i < nodeList.getLength(); i++) {
+            Node node = nodeList.item(i);
+
+            if(node.getNodeType() != Node.ELEMENT_NODE){
+                throw new Exception("Removal failed: XPath must return element types.");
+            }
+
+            if (attributeName == null || attributeName.isEmpty()) {
+                node.getParentNode().removeChild(node);
+            }
+            else{
+                node.getAttributes().removeNamedItem(attributeName);
+            }
+        }
+    }
+
+    public static void setValueToList(NodeList nodeList, String attributeName, String value) throws Exception{
+        for (int i = 0; i < nodeList.getLength(); i++) {
+            Node node = nodeList.item(i);
+
+            if(node.getNodeType() != Node.ELEMENT_NODE){
+                throw new Exception("Removal failed: XPath must return element types.");
+            }
+
+            if(attributeName == null || attributeName.isEmpty()) {
+                node.setTextContent(value);
+            }
+            else {
+                ((Element) node).setAttribute(attributeName, value);
+            }
+        }
+    }
+
+    public static void validateAgainstXsd(String xmlDocument, String xsdDocument) throws Exception{
+        SchemaFactory schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+        Schema schema = schemaFactory.newSchema(new StreamSource(new StringReader(xsdDocument)));
+        Validator validator = schema.newValidator();
+        validator.validate(new StreamSource(new StringReader(xmlDocument)));
+    }
+
+    public static String xPathQuery(Document doc, XPathExpression expr, String queryType, String delimiter) throws Exception{
+        if(queryType.equals(Constants.QueryTypes.NODE_LIST)) {
+            return xPathNodeListQuery(doc, expr, delimiter);
+        }
+        else if (queryType.equals(Constants.QueryTypes.NODE)) {
+            return xPathNodeQuery(doc, expr);
+        }
+        else if (queryType.equals(Constants.QueryTypes.VALUE)) {
+            return xPathValueQuery(doc, expr);
+        }
+        else{
+            throw new Exception("Invalid query type");
+        }
+    }
+
+    public static String xPathNodeListQuery(Document doc, XPathExpression expr, String delimiter) throws Exception {
+        NodeList nodeList = (NodeList) expr.evaluate(doc, XPathConstants.NODESET);
+        return nodeListToString(nodeList,delimiter);
+    }
+
+    public static String xPathNodeQuery(Document doc, XPathExpression expr) throws Exception {
+        Node n = (Node) expr.evaluate(doc, XPathConstants.NODE);
+        return nodeToString(n);
+    }
+
+    public static String xPathValueQuery(Document doc, XPathExpression expr) throws Exception {
```

Comment:
methods like this that are only used inside this class should be private, please check other methods(find Usages alt+F7)